src/main/scala/com/danielasfregola/twitter4s/http/clients/streaming/StreamingClientNew.scala

```diff
@@ -0,0 +1,76 @@
+package com.danielasfregola.twitter4s.http.clients.streaming
+
+import akka.NotUsed
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.{HttpRequest, HttpResponse}
+import akka.stream.scaladsl.{Framing, Source}
+import akka.stream.{ActorMaterializer, SharedKillSwitch}
+import akka.util.ByteString
+import com.danielasfregola.twitter4s.entities.streaming.StreamingMessage
+import com.danielasfregola.twitter4s.entities.{AccessToken, ConsumerToken}
+import com.danielasfregola.twitter4s.exceptions.TwitterException
+import com.danielasfregola.twitter4s.http.clients.OAuthClient
+import com.danielasfregola.twitter4s.http.oauth.OAuth1Provider
+import org.json4s.native.Serialization
+
+import scala.concurrent.Future
+import scala.util.Try
+
+private[twitter4s] class StreamingClientNew(val consumerToken: ConsumerToken, val accessToken: AccessToken)(
+    implicit val system: ActorSystem)
+    extends OAuthClient {
+
+  override def withLogRequest: Boolean = true
+  override def withLogRequestResponse: Boolean = true
+
+  implicit val materializer = ActorMaterializer()
+  implicit val ec = materializer.executionContext
+
+  def preProcessing(): Unit = ()
+
+  lazy val oauthProvider = new OAuth1Provider(consumerToken, Some(accessToken))
+
+  def getStream(request: HttpRequest, killSwitch: SharedKillSwitch): Future[Source[StreamingMessage, NotUsed]] = {
+    withOAuthHeader(None)(materializer)(request).map(_ => requestIntoStream(killSwitch)(request))
+  }
+
+  private def requestIntoStream(killSwitch: SharedKillSwitch)(implicit request: HttpRequest) = {
+    Source
+      .single(request)
+      .via(connection)
+      .flatMapConcat(response => processHttpResponse(response, killSwitch))
+  }
+
+  def processHttpResponse(httpResponse: HttpResponse, killSwitch: SharedKillSwitch) = {
+    httpResponse match {
+      case response if response.status.isSuccess =>
+        processBody(response, killSwitch)
+      case failureResponse =>
+        val statusCode = failureResponse.status
+        val msg = "Stream could not be opened"
+        parseFailedResponse(failureResponse).map(ex => logger.error(s"$msg: $ex"))
+        Source.failed(TwitterException(statusCode, s"$msg. Check the logs for more details"))
+    }
+  }
+
+  def processBody[T: Manifest, A](
```

Comment:
Type parameter A passed but not used? Also, I am confused by the `Any` in the return type `Source[StreamingMessage, Any]`. Could you please clarify your reasoning?

Reply:
Lots of type information was not updated by me in this PR, it was leftover from your implementation